gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -565,16 +527,19 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
             }
             // No else; The cases should be exhaustive
         }
-        kspGeneratedSourceSet.kotlin.srcDir(project.files(kotlinOutputDir, javaOutputDir).builtBy(kspTaskProvider))
+
+        val generatedSources = arrayOf(
+            project.files(kotlinOutputDir).builtBy(kspTaskProvider),
+            project.files(javaOutputDir).builtBy(kspTaskProvider),
+        )
         if (kotlinCompilation is KotlinCommonCompilation) {
-            // Do not make common source sets depend on generated source sets.
-            // They will be observed by downstreams and confuse processors.
-            kotlinCompileProvider.configure {
-                it.source(kspGeneratedSourceSet.kotlin)
-            }
+            // Do not add generated sources to common source sets.
+            // They will be observed by downstreams and violate current build scheme.
+            kotlinCompileProvider.configure { it.source(*generatedSources) }
         } else {
-            kotlinCompilation.defaultSourceSet.dependsOn(kspGeneratedSourceSet)
+            kotlinCompilation.defaultSourceSet.kotlin.srcDirs(*generatedSources)
```

Comment:
Let's look into adding an API in KGP that allows:
- registering source-generating tasks
- getting all sources (without the sources you've just generated)
?

Reply:
That'll be awesome. Before the API is available in KGP, I guess I'll have to choose between:
1. this PR, that needs to iterate through all sources, or
2. give up IDE support

Whether making generated sources available to other Gradle tasks is also a difference.

Reply:
Can you please file a YT issue and I'll raise this with JB?